lightning/src/ln/onion_utils.rs

```diff
@@ -425,12 +425,24 @@ pub(super) fn construct_onion_packet(
 #[allow(unused)]
 pub(super) fn construct_trampoline_onion_packet(
 	payloads: Vec<msgs::OutboundTrampolinePayload>, onion_keys: Vec<OnionKeys>,
-	prng_seed: [u8; 32], associated_data: &PaymentHash, length: u16,
+	prng_seed: [u8; 32], associated_data: &PaymentHash, length: Option<u16>,
 ) -> Result<msgs::TrampolineOnionPacket, ()> {
-	let mut packet_data = vec![0u8; length as usize];
+	let minimum_packet_length: usize = payloads.iter().map(|p| p.serialized_length() + 32).sum();
```

Comment:
Nit: doesn't seem like `: usize` is needed, but I let you make the decision on style - from my point of view, we are talking about lengths here, so not surprising that the type is `usize`.